OpenRA.Game/Widgets/Widget.cs

```diff
@@ -162,6 +173,10 @@ public abstract class Widget
 		public bool Visible = true;
 		public bool IgnoreMouseOver;
 		public bool IgnoreChildMouseOver;
+		public Color DebugBoundsColor = Color.Red;
+		public float DebugBoundsWidth = 1;
+		public Color DebugMouseOverBoundsColor = Color.Green;
+		public float DebugMouseOverBoundsWidth = 2;
```

Comment:
I got to this point but realized that I'll be highlighting all children unless I pass two floats and two colors, which is the route we could take.

```diff
diff --git a/OpenRA.Game/Widgets/RootWidget.cs b/OpenRA.Game/Widgets/RootWidget.cs
index dad5b21..13177d3 100644
--- a/OpenRA.Game/Widgets/RootWidget.cs
+++ b/OpenRA.Game/Widgets/RootWidget.cs
@@ -8,15 +8,37 @@
  */
 #endregion
 
+using System.Drawing;
 using OpenRA.Graphics;
 
 namespace OpenRA.Widgets
 {
 	public class RootWidget : ContainerWidget
 	{
+		public Color DebugBoundsColor = Color.Red;
+		public float DebugBoundsWidth = 1;
+		public Color DebugMouseOverBoundsColor = Color.Green;
+		public float DebugMouseOverBoundsWidth = 2;
+
 		public RootWidget()
 		{
 			IgnoreMouseOver = true;
+
+			Color debugBoundsColor;
+			if (ChromeMetrics.TryGet<Color>("DebugBoundsColor", out debugBoundsColor))
+				DebugBoundsColor = debugBoundsColor;
+
+			float debugBoundsWidth;
+			if (ChromeMetrics.TryGet<float>("DebugBoundsWidth", out debugBoundsWidth))
+				DebugBoundsWidth = debugBoundsWidth;
+
+			Color debugMouseOverBoundsColor;
+			if (ChromeMetrics.TryGet<Color>("DebugMouseOverBoundsColor", out debugMouseOverBoundsColor))
+				DebugMouseOverBoundsColor = debugMouseOverBoundsColor;
+
+			float debugMouseOverBoundsWidth;
+			if (ChromeMetrics.TryGet<float>("DebugMouseOverBoundsWidth", out debugMouseOverBoundsWidth))
+				DebugMouseOverBoundsWidth = debugMouseOverBoundsWidth;
 		}
 
 		public override bool HandleKeyPress(KeyInput e)
diff --git a/OpenRA.Game/Widgets/Widget.cs b/OpenRA.Game/Widgets/Widget.cs
index cbad4e7..62d74e9 100644
--- a/OpenRA.Game/Widgets/Widget.cs
+++ b/OpenRA.Game/Widgets/Widget.cs
@@ -173,10 +173,6 @@ public abstract class Widget
 		public bool Visible = true;
 		public bool IgnoreMouseOver;
 		public bool IgnoreChildMouseOver;
-		public Color DebugBoundsColor = Color.Red;
-		public float DebugBoundsWidth = 1;
-		public Color DebugMouseOverBoundsColor = Color.Green;
-		public float DebugMouseOverBoundsWidth = 2;
 
 		// Calculated internally
 		public Rectangle Bounds;
@@ -201,11 +197,6 @@ public Widget(Widget widget)
 			IgnoreChildMouseOver = widget.IgnoreChildMouseOver;
 			IgnoreMouseOver = widget.IgnoreMouseOver;
 
-			DebugBoundsColor = widget.DebugBoundsColor;
-			DebugBoundsWidth = widget.DebugBoundsWidth;
-			DebugMouseOverBoundsColor = widget.DebugMouseOverBoundsColor;
-			DebugMouseOverBoundsWidth = widget.DebugMouseOverBoundsWidth;
-
 			foreach (var child in widget.Children)
 				AddChild(child.Clone());
 		}
@@ -439,7 +430,7 @@ public virtual void PrepareRenderablesOuter()
 
 		public virtual void Draw() { }
 
-		public void DrawDebugRenderBounds(bool drawEvenIfInvisible = false, bool recurseDownAllChildren = true)
+		public void DrawDebugRenderBounds(float width, Color color, bool drawEvenIfInvisible = false, bool recurseDownAllChildren = true)
 		{
 			if (drawEvenIfInvisible || IsVisible())
 			{
@@ -447,12 +438,12 @@ public void DrawDebugRenderBounds(bool drawEvenIfInvisible = false, bool recurse
 				Game.Renderer.RgbaColorRenderer.DrawRect(
 					new float2(RenderBounds.Left, RenderBounds.Top),
 					new float2(RenderBounds.Right, RenderBounds.Bottom),
-					isMouseOverWidget ? DebugMouseOverBoundsWidth : DebugBoundsWidth,
-					isMouseOverWidget ? DebugMouseOverBoundsColor : DebugBoundsColor);
+					width,
+					color);
 
 				if (recurseDownAllChildren)
 					foreach (var child in Children.OfType<Widget>().Reverse())
-						child.DrawDebugRenderBounds();
+						child.DrawDebugRenderBounds(/* what do I pass here? */);
 			}
 		}
```